authors/apps/like_dislike/views.py

```diff
@@ -16,26 +16,39 @@
 from .models import LikeDislike
 from .renderers import LikeDislikeJSONRenderer
 from .serializers import PreferenceSerializer
+from authors.apps.comments.models import Comments
+from authors.apps.authentication.utils import status_codes, swagger_body
 
 
 class PreferenceView(generics.GenericAPIView):
+    """Preference view"""
 
     permission_classes = (IsAuthenticated,)
     renderer_classes = (LikeDislikeJSONRenderer,)
     serializer_class = PreferenceSerializer
     pref = None  # Preference type Like/Dislike
+    model = None  # Model (Articles/Comments)
     swagger_schema = SwaggerAutoSchema
 
-    def post(self, request, slug):
-        article = Article.objects.get(slug=slug)
-        pref_status = None
+    @swagger_auto_schema(
+        request_body=swagger_body(None),
+        responses=status_codes(codes=(200, 401))
+    )
+    def post(self, request, slug, pk=None):
```

Comment:
add comments helps us understand what is happening

Reply:
Cool, thanks,,,let me do that